src/HeapDump/GCHeapDumper.cs

```diff
@@ -391,6 +392,12 @@ private DataTarget InitializeClrRuntime(string processDumpFile, int processID, o
     /// </summary>
     public ulong PromotedBytesThreshold;
 
+
+    /// <summary>
+    /// The token credential to use for symbol server authentication.
+    /// </summary>
+    public TokenCredential SymbolsAuthTokenCredential;
```

Comment:
Seems strange to have both a public constructor arg AND a public settable field. My preferred approach if you need to provide this for consumption would be (in order of preference):
* An internal get-only property
* A public get-only property (if it really needs to be public)

Reply:
public fields are typically thought of as an anti-pattern

Reply:
You're right - The constructor really doesn't need the argument anymore, especially with a default `NULL` value.  I've removed it.  One thing to note is that the `GCHeapDumper` class is part of HeapDump which isn't publicly shipped as surface area that folks can depend upon.  It's really an internal implementation detail of PerfView, and so I imagine that's why it uses public fields.  I can change this if you're feeling strongly about this, but for the purposes of this change, I'm just following the existing pattern.